### PR TITLE
small pr to make firemode code easier | or, how firemode code was so fucking terrible that this pr touches like 20-30 files

### PR DESCRIPTION
**Pull request by rye-rice**
## About The Pull Request

Wanted to add pride flags for pride month, but had to get this 'small pr' out the way. Oops.

tldr; as a player, this should not affect you. Like at all. Move along

This changes firemodes to not be fucking terrible codewise and to touch as a developer. Yes, i HAD to touch gun.dm that much. That was how deep rooted the rot was.

Also get's rid of the 0.4 second hardcoded fire delay, now pistols and revolvers shoot as fast as they were intended. Great. This meant that i had to touch a bunch of numbers i set to 0 out of frustration at it not working, being completely unaware of the hardcoded fire delay.

This also fixes the E-40 to have laser full auto and to have an ammo counter. I wasn't even trying to fix the E-40, but by unshittifying firemode code i got it to work. Wow.

## Why It's Good For The Game

This should make the 3 upcoming gun expansions less ass to add.

## Changelog

:cl:
add: Changing firemodes on guns now have new sprites
fix: The E-40 now works as intended!
add: The E-40 now has an ammo counter
balance: The P16 shoots slightly faster.
balance: .299 Eoehoma caseless has gotten a slight nerf, while the E-40 shoots slightly faster.
balance: Two E-40s max are obtainable from the black market should it spawn. Blank market price cap of the E-40 has also increased

/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -27,25 +26,55 @@
...
+	///Firemode index, due to code shit this is the currently selected firemode
+	var/firemode_index
+	/// Our firemodes, subtract and add to this list as needed. NOTE that the autofire component is given on init when FIREMODE_FULLAUTO is here.
+	var/list/gun_firemodes = list(FIREMODE_SEMIAUTO, FIREMODE_BURST, FIREMODE_FULLAUTO, FIREMODE_OTHER, FIREMODE_OTHER_TWO)
+	/// A acoc list that determines the names of firemodes. Use if you wanna be weird and set the name of say, FIREMODE_OTHER to "Underbarrel grenade launcher" for example.
+	var/list/gun_firenames = list(FIREMODE_SEMIAUTO = "single", FIREMODE_BURST = "burst fire", FIREMODE_FULLAUTO = "full auto", FIREMODE_OTHER = "misc. fire", FIREMODE_OTHER_TWO = "very misc. fire")
+	///BASICALLY: the little button you select firing modes from? this is jsut the prefix of the icon state of that. For example, if we set it as "laser", the fire select will use "laser_single" and so on.
+	var/fire_select_icon_state_prefix = ""
+	///If true, we put "safety_" before fire_select_icon_state_prefix's prefix. ex. "safety_laser_single"
+	var/adjust_fire_select_icon_state_on_safety = FALSE
```
Comment (FalloutFalcon) [inline code/modules/projectiles/gun.dm:63]: this is a wordy var but shrug

**Review by FalloutFalcon — COMMENTED** [review]
will review properly at some point,, gun.dm is annyoing to read cause of the shuffled procs..
on first skim looks good
would be nice to see all the random balance changes as its own pr..

**Review by Sun-Soaked — APPROVED** [review]
Worked in testmerge, superficial look seems good. 

it's on me if this breaks somehow
